### PR TITLE
[FIX-#1740] MzTabExporter: modification "Phospho (ST)" throws error

### PR DESCRIPTION
**Pull request by PetraGutenbrunner**
[FIX-#1740] Split Phospho (ST) to Phospho (S), Phospho (T)
When Mascot output XML parsed, split the fixed modifications immediately after parsing, because identifier information don't needs to be preserved.

The order of variable modifications needs to be preserved, because the position of the modification in the list indicates the index of these pattern:
pep_seq: QKAAGSK
pep_var_mod_pos: 4.0000000.0
=> modification on the 4th position is the n-terminal modification.

mascot_xml_mod_definition

Fixed in addition Mascot XML Parser - Terminal Modifications:

[FIX 1] Mascot adds fixed modifications, which cannot be set, additionally as variable modification.
Error example: Fixed mod: "Gln->pyro-Glu (N-term Q)"
<warning number="1">&apos;Gln-&gt;pyro-Glu (N-term Q)&apos; can only be used as a variable modification; setting it to variable</warning>

mascot_fixed_added_as_var_mod

Solution: Parsing the warnings of Mascot:
<warning number="0">&apos;Ammonia-loss (N-term C)&apos; can only be used as a variable modification; setting it to variable</warning>
<warning number="1">&apos;Gln-&gt;pyro-Glu (N-term Q)&apos; can only be used as a variable modification; setting it to variable</warning>

If mentioned modifications occurs as fixed modification --> remove it. 

[FIX 2] Fixed modifications: N-term/C-term modifications on a specific amino acid are ignored.
If a terminal modification is set in the search parameters as fixed and it is defined for a specific amino acid, the specifity is ignored and
the terminal modification is set on every sequence in the Mascot idXML output.
Reason: in the mascot XML parsing step, amino acid specifications are not considered, when the fixed modifications are added to the sequence.

Set in all peptides as N-term modification:
fixed_term_with_specifity_set_in_all_psms

Solution: When set the fixed modifications, specific amino acid definitions are considered now

modified:   ../src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h
modified:   ../src/openms/include/OpenMS/FORMAT/HANDLERS/MascotXMLHandler.h
modified:   ../src/openms/source/CHEMISTRY/ModificationsDB.cpp
modified:   ../src/openms/source/FORMAT/HANDLERS/MascotXMLHandler.cpp
modified:   ../src/tests/class_tests/openms/data/MascotXMLFile_test_out_3.idXML
modified:   ../src/tests/class_tests/openms/source/MascotXMLFile_test.cpp
modified:   ../src/tests/topp/IDFileConverter_1_output.idXML
modified:   ../src/tests/topp/IDFileConverter_5_output.idXML

### CODE REVIEW

File: src/openms/source/CHEMISTRY/ModificationsDB.cpp [hunk 1]
```diff
@@ -199,6 +199,11 @@ namespace OpenMS
     }
     return **mods.begin();
   }
+  
+  bool ModificationsDB::has(String modification) const
+  {
+    return (modification_names_.has(modification)) ? true : false;
```
Comment (timosachsenberg) [inline src/openms/source/CHEMISTRY/ModificationsDB.cpp]: return modification_names_.has(modification) should be enough

File: src/tests/class_tests/openms/data/MascotXMLFile_test_out_3.idXML [hunk 2]
```diff
@@ -4,7 +4,8 @@
 	<SearchParameters id="SP_0" db="p996_silac" db_version="fgcz_ipi_human_v3.68_decoy_contaminants_20120103.fasta" taxonomy="All entries" mass_type="monoisotopic" charges="" enzyme="trypsin" missed_cleavages="2" precursor_peak_tolerance="20" precursor_peak_tolerance_ppm="true" peak_mass_tolerance="0.02" peak_mass_tolerance_ppm="false" >
 		<FixedModification name="Carbamidomethyl (C)" />
 		<VariableModification name="Oxidation (M)" />
-		<VariableModification name="Deamidated (NQ)" />
+		<VariableModification name="Deamidated (N)" />
+    <VariableModification name="Deamidated (Q)" />
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/data/MascotXMLFile_test_out_3.idXML]: maybe a whitespace issue here

File: src/tests/topp/IDFileConverter_1_output.idXML [hunk 3]
```diff
@@ -3,7 +3,8 @@
 <IdXML version="1.3" xsi:noNamespaceSchemaLocation="http://open-ms.sourceforge.net/SCHEMAS/IdXML_1_3.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 	<SearchParameters id="SP_0" db="UPEcoli_SpikeIn" db_version="UPEcoli_SpikeIn_Jun13.fasta" taxonomy="All entries" mass_type="monoisotopic" charges="" enzyme="trypsin" missed_cleavages="1" precursor_peak_tolerance="10" precursor_peak_tolerance_ppm="true" peak_mass_tolerance="0.5" peak_mass_tolerance_ppm="false" >
 		<FixedModification name="Carbamidomethyl (C)" />
-		<VariableModification name="Deamidated (NQ)" />
+		<VariableModification name="Deamidated (N)" />
+    <VariableModification name="Deamidated (Q)" />
```
Comment (timosachsenberg) [inline src/tests/topp/IDFileConverter_1_output.idXML]: here too: tab vs. space?

File: src/openms/source/CHEMISTRY/ModificationsDB.cpp (hunk 1 above)
Comment (PetraGutenbrunner) [inline src/openms/source/CHEMISTRY/ModificationsDB.cpp]: oh dear, that has never happend :) thx

**Comment by timosachsenberg** [comment]
+1

**Pull request merged**